phf_codegen/src/lib.rs

```diff
@@ -147,47 +159,55 @@ impl<K: Hash+PhfHash+Eq+fmt::Debug> Map<K> {
 
         let state = phf_generator::generate_hash(&self.keys);
```

Comment:
I think the advantage of returning a struct that implements `Display` is that you can calculate this just once in case the user needs to print this multiple times for some reason.

Reply:
And also, given how intensive this operation is, I don't know if it should be hidden inside a `Display` impl.